### DISCUSSION TITLE
Functions path in vercel.json unclear on RemixJs

### DISCUSSION THREAD

**Discussion by Edran0111** [vercel/community#5749] [Help] [answered]
### Summary

Hello everyone !
I'm trying to build my application to increase the maxDuration and memory of my project because it fails otherwise on each routes(i have the pro plan).
What I'm trying to to is simple i just want to know the path of the source build to put in the vercel.json : 
 ```
 {
     "functions":{
      "api/build/*.ts": {
          "memory": 3008,
          "maxDuration": 30
      }
}
```

But the problem is this one : 
`Error: The pattern "api/build/*.ts" defined in `functions` doesn't match any Serverless Functions inside the `api` directory.`

I do not know what to put in the specification of the files (here => "api/build/*.ts"). I've tried to do all the possibilities, but I could not find a clue on the documentation regarding the path that i'm supposed to put especially on a RemixJs framework. Honestly I'd be glad just to find what would the repository would look like!

### Steps to Reproduce

I have found some questions that put the build Source and Output so here they are : 
Here's what my folder looks like a clear Remix app structure:
![image](https://github.com/vercel/community/assets/82849327/327d09e5-0a30-4558-8000-0eb5fcc8e74a)
And here is what my folder looks like in the output
![image](https://github.com/vercel/community/assets/82849327/33378ebd-9d1e-4a68-991f-a88d4b3628d8)

So please if someone has the answer to that question because honestly I'm a little lost and I know the answer is here but i can't seem to find it.

PS: I know i can hardcode it into each routes, but I think it can be good to answer this unasked question on the internet as a reference.

**Comment by leerob** [comment, marked as answer]
We recommend using the route level option for durations:

```js
// This function can run for a maximum of 5 seconds
export const config = {
  maxDuration: 5,
};
 
export default function Serverless() {
  return (
    <div style={{ fontFamily: 'system-ui, sans-serif', lineHeight: '1.4' }}>
      <h1>Welcome to Remix with Vercel (Serverless)</h1>
    </div>
  );
}
```

https://vercel.com/docs/frameworks/remix

**Reply by leerob** [reply]
For memory, you are correct that currently you would need to use `vercel.json`. Did you try using the actual routes there? e.g. `app/routes/*.ts`.

**Reply by Edran0111** [reply]
Thanks for your answer, the thing that is currently working is by specifying each route with your recommandation: 

```
export const config = {
  memory:2048,
  maxDuration: 300,
};
```

Indeed i have tried a lot but the building failed using these paths in the vercel.json : 
```
-app/routes/*.ts 
-app/routes/**/*.ts
-build/index.js
-api/*.js
-app/api/**/*.ts
-*.js
-build/routes/*.js
-api/build/routes/*.js
-*.ts
-app/api/**/*.ts
-build/routes/*.js
-public/build/routes/*.js
-api/build/routes/*.js
-api/*.js
-build/*.js
-routes/*.js
-api/**/*.ts
-api/*.ts
-build/index.js
```
Thank you for your answer !

**Comment by felquis** [comment]
I'm having the same problem, but when I try the accepted answer it cause the endpoint to return 404

**Reply by felquis** [reply]
This is exactly what's happening https://github.com/orgs/vercel/discussions/4244

**Reply by felquis** [reply]
I've figure I'd need remix v2 new routing convention to make it work with `export const config`
https://remix.run/docs/en/main/start/v2#upgrading-to-the-new-convention just rewrote all app routes
it was worth it as now the webapp is running with remix latest version
